pkg/generators/forwarding/fluentd/templates.go

```diff
@@ -568,17 +568,28 @@ tls_cert_path {{ .SecretPath "ca-bundle.crt"}}
   @type file
   path '{{.BufferPath}}'
   queued_chunks_limit_size "#{ENV['BUFFER_QUEUE_LIMIT'] || '1024' }"
-  chunk_limit_size "#{ENV['BUFFER_SIZE_LIMIT'] || '1m' }"
+  {{ if .TotalLimitSize }}
```

Comment:
@alanconway mentioned this to me while writing the proposal and I believe it has merit.  We should consider passing the changes in via env vars as it would significantly simplify the config generation and reduce the number of:
* if/else checks
* helper methods needed
I like the idea of being explicit in config but in the interest of mentally embracing tuning it may make more sense

Reply:
TLDR: do what's easiest :) My thinking was that as long as we have the env vars in run.sh anyway, we might as well take advantage of them. In future it might be better to simplify run.sh and get rid of those env-vars so there's only one place where we set the config - but we can do that when we get there, there's no benefit in anticipating it before we have actually simplified run.sh, which isn't for this PR.

Reply:
We concluded to go with the present approach using template helper methods.